Applications/FileIO/XmlIO/Qt/XmlPrjInterface.cpp

```diff
@@ -0,0 +1,507 @@
+                 base_obj_type != DataHolderLib::BaseObjType::MESH)
+            obj_name = content.toStdString();
+        else if (node_name == "type")
+            type = T::convertStringToType(content.toStdString());
+        else if (node_name == "mesh")
+        {
+            base_obj_type = DataHolderLib::BaseObjType::MESH;
+            base_obj_name = content.toStdString();
+            obj_name.clear();
+        }
+        else if (node_name == "field_name")
+            param_name = content.toStdString();
+        else if (node_name == "parameter")
+        {
+            QDomNode val = findParam(param_root, content);
+            if (val == QDomNode())
+                continue;
+            param_name = content.toStdString();
+            param_node = nodeList.at(i);
+        }
+    }
+
+    if (!param_name.empty())
+    {
+        T* cond = new T(pvar, param_name, type);
+        if (base_obj_type == DataHolderLib::BaseObjType::MESH)
+            cond->setMesh(base_obj_name);
+        else
+            cond->setGeoObject(base_obj_name, obj_name);
+
+        return cond;
+    }
+    return new T({"", 0, 0}, "", T::NONE);
+}
+
+int XmlPrjInterface::writeToFile(const std::string& filename)
+{
+    _filename = filename;
+    return BaseLib::IO::Writer::writeToFile(filename);
+}
+
+bool XmlPrjInterface::write()
+{
+    GeoLib::GEOObjects& geo_objects = _project.getGEOObjects();
+    QFileInfo fi(QString::fromStdString(_filename));
+    std::string path((fi.absolutePath()).toStdString() + "/");
+
+    _out << "<?xml version=\"1.0\" encoding=\"ISO-8859-1\"?>\n";  // xml
+                                                                  // definition
+    _out << "<?xml-stylesheet type=\"text/xsl\" "
+            "href=\"OpenGeoSysProject.xsl\"?>\n\n";  // stylefile definition
+
+    QDomDocument doc("OGS-PROJECT-DOM");
+    QDomElement root = doc.createElement("OpenGeoSysProject");
+    root.setAttribute("xmlns:ogs", "http://www.opengeosys.org");
+    root.setAttribute("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance");
+    root.setAttribute(
+        "xsi:noNamespaceSchemaLocation",
+        "http://www.opengeosys.org/images/xsd/OpenGeoSysProject.xsd");
+
+    doc.appendChild(root);
+
+    // meshes
+    auto const& mesh_vector = _project.getMeshObjects();
+    for (auto const& mesh : mesh_vector)
+    {
+        // write mesh file
+        MeshLib::IO::writeMeshToFile(*mesh, path + mesh->getName() + ".vtu");
+
+        // write entry in project file
+        QDomElement mesh_tag = doc.createElement("mesh");
+        root.appendChild(mesh_tag);
+        QDomText filename_text =
+            doc.createTextNode(QString::fromStdString(mesh->getName()));
+        mesh_tag.appendChild(filename_text);
+    }
+
+    // geometries
+    std::vector<std::string> geo_names;
+    geo_objects.getGeometryNames(geo_names);
+    for (std::vector<std::string>::const_iterator it(geo_names.begin());
+         it != geo_names.end();
+         ++it)
```

Comment:
can we get rid of those iterators, please! This is so c++03 style.
There are for-range-loops, which are readable.